platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFilePath.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.signed;
+
+import static com.swirlds.common.io.utility.FileUtils.getAbsolutePath;
+import static com.swirlds.logging.legacy.LogMarker.EXCEPTION;
+import static com.swirlds.platform.state.signed.SignedStateFileUtils.SIGNED_STATE_FILE_NAME;
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.isDirectory;
+
+import com.swirlds.common.config.StateConfig;
+import com.swirlds.common.system.NodeId;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Utility methods for determining the path of signed states on disk.
+ */
+public class SignedStateFilePath {
+    private static final Logger logger = LogManager.getLogger(SignedStateFilePath.class);
+    final StateConfig stateConfig;
+
+    /**
+     * Create a new instance of this class.
+     *
+     * @param stateConfig the config that contains the location of the saved state directory
+     */
+    public SignedStateFilePath(@NonNull final StateConfig stateConfig) {
+        this.stateConfig = stateConfig;
+    }
```

Comment:
It seems odd to me that this class can be instantiated. All the instance is used for is to store a parameter value that is used by the utility methods. For stuff like this, I think it would make sense for each method to be static and to take a configuration object as a parameter.

Reply:
in the future, i plan to put more parameters in there, like `mainClassName`, `platformId` & `swirldName`. this would provide with more decoupling of the path resolution and the user of this utility. the user does not need to care what this path is based on, it just says "give me the path". this would also make unit testing simpler, since we can replace this implementation with a test one in a unit test. I think it makes less sense to make it static and constantly pass it a bunch of parameters that it needs.

Reply:
#10292